frontend/__tests__/components/AddBookModal/add-book-modal-snapshot.test.js

```diff
@@ -0,0 +1,22 @@
+import React from "react";
+import { renderer } from "../../../test-utils";
+import AddBookModal from "../../../components/AddBookModal/AddBookModal";
+
+jest.mock("react-dom", () => {
+  const originalModule = jest.requireActual("react-dom");
+
+  return {
+    ...originalModule,
+    createPortal: (element) => element,
+  };
+});
+
+describe("AddBookModal Snapshot", () => {
+  const spy = jest.spyOn(document, "getElementById");
+  spy.mockReturnValue(<div id="app-root" />);
```

Comment:
Please elaborate on this. Once again, we would prefer you not use snapshot testing. This brings up the point though that with code like this that looks odd, you should be able to explain why this is required and the technical implementation of what this does.